api/office_structure/schema.py

```diff
@@ -0,0 +1,49 @@
+import graphene
+from graphene_sqlalchemy import SQLAlchemyObjectType
+
+from api.office_structure.models import OfficeStructure as OfficeStructureModel
+from helpers.auth.error_handler import SaveContextManager
+from helpers.auth.authentication import Auth
+from utilities.validations import (
+    validate_empty_fields,
+    validate_parent_node_id,
+    )
+
+
+class OfficeStructure(SQLAlchemyObjectType):
+    """Autogenerated Return type for OfficeStructure"""
+
+    class Meta:
+        model = OfficeStructureModel
+
+
+class CreateNode(graphene.Mutation):
+    """ Returns payload on node creation"""
+
+    class Arguments:
+        name = graphene.String(required=True)
+        parent_id = graphene.Int(required=False)
+    node = graphene.Field(OfficeStructure)
+
+    @Auth.user_roles('Admin')
+    def mutate(self, info, **kwargs):
```

Comment:
Great work but currently I can create a node with an empty space, maybe you can look at the validate_empty_fields() and see how you can strip the spaces
<img width="1188" alt="screenshot 2019-03-05 at 17 28 32" src="https://user-images.githubusercontent.com/23398223/53812352-5d8c3b80-3f6c-11e9-98d0-a06469e00bcf.png">

Reply:
This means, I need to refactor `validate empty fields`. Thanks for the pointer let me work on it. I guess this is a bug to fix on V1 too. cc @Bonifase , cc @joshuaocero

Reply:
This `https://www.pivotaltracker.com/story/show/164361872` on the PT board would solve the issue. Person working on that task can use the same solution I have used here for v1 only, v2 will use this once this PR is merged. This has been solved @michael-basweti